dongkyu/initial-settings-express/myapp/app.js

```diff
@@ -112,10 +112,70 @@ app.get("/posts", async (req, res) => {
   })
 
 
+  app.patch("/posts/:postId", async (req, res) => {
+    
+    const { postId } = req.params; 
+
+    const { title, content, userId} = req.body
+     
+    await appDataSource.query(
+      `UPDATE posts
+        SET 
+          title = ?,
+          content = ?
+        WHERE user_id = ?
+      `,
+      [ title, content, userId ]
+   );
+  
+    await appDataSource.query(
+      `SELECT
+        users.id as userId,
+        users.name as userName,
+        posts.id as postingId,
+        posts.title as postingTitle,
+        posts.content as postingContent
+      FROM users
+      JOIN posts
+      ON users.id = posts.user_id AND posts.id = ${postId}
+      `,
+      (err, rows) => {
+      res.status(201).json({data: rows})
+  });
+  })
+
+
+  app.delete('/posts/:postId', async (req, res) => {
+    const { postId } = req.params;
+    await appDataSource.query(
+      `DELETE FROM posts
+        WHERE posts.id = ${postId}
+        `
+    )
+    res.status(200).json({ message: 'postingDeleted' });
+  });
+  
+   
+  app.post("/likes", async (req, res) => {
+    const {userId, postId} = req.body
+
+    await appDataSource.query(
+      `INSERT INTO likes (
+        user_id,
+        post_id
+      ) VALUES (?, ?);
+      `,
+      [userId, postId]
+    );
+    res.status(201).json({ message : "likeCreated"})
+  })
```

Comment:
- 좋아요 기능을 구현할 때에는 주의해야할 점이 있습니다.
- 하나의 사용자가 하나의 게시물에 대해서 좋아요를 할 수 있는 횟수는 단 한 번 이어야 합니다.
- 지금 구조에서는 좋아요를 누를 때마다 아래와 같이 likes 테이블에 계속해서 중복으로 데이터가 저장됩니다.
```suggestion
+----+---------+---------+---------------------+
| id | user_id | post_id | created_at          |
+----+---------+---------+---------------------+
|  1 |       1 |       3 | 2022-07-16 14:18:26 |
|  2 |       1 |       3 | 2022-07-16 14:18:26 |
|  3 |       1 |       3 | 2022-07-16 14:18:26 |
|  4 |       1 |       3 | 2022-07-16 14:18:26 |
|  5 |       1 |       3 | 2022-07-16 14:18:47 |
|  6 |       1 |       3 | 2022-07-16 14:18:47 |
+----+---------+---------+---------------------+
```

이를 막기 위해서, likes 테이블에 user_id와 post_id를 기준으로 UNIQUE 제약조건을 추가하는 migration 파일을 추가로 만들어서 올려주세요.